packages/flutter/lib/src/services/scribe.dart

```diff
@@ -0,0 +1,135 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'message_codec.dart';
+import 'system_channels.dart';
+
+/// An interface into Android's stylus handwriting text input.
+///
+/// Allows handwriting directly on top of a text input using a stylus.
+///
+/// See also:
+///
+///  * [EditableText.stylusHandwritingEnabled], which controls whether Flutter's
+///    built-in text fields support handwriting input.
+///  * [SystemChannels.scribe], which is the [MethodChannel] used by this
+///    class, and which has a list of the methods that this class handles.
+abstract final class Scribe {
+  static const MethodChannel _channel = SystemChannels.scribe;
+
+  /// A convenience method to check if the device currently supports Scribe
+  /// stylus handwriting input.
+  ///
+  /// Call this each time before calling [startStylusHandwriting] to make sure
+  /// it's available.
+  ///
+  /// {@tool snippet}
+  /// This example shows using [isFeatureAvailable] to confirm that
+  /// [startStylusHandwriting] can be called.
+  ///
+  /// ```dart
+  /// if (!await Scribe.isFeatureAvailable()) {
+  ///   // The device doesn't support stylus input right now, or maybe at all.
+  ///   return;
+  /// }
+  ///
+  /// // Scribe is supported, so start it.
+  /// Scribe.startStylusHandwriting();
+  /// ```
+  /// {@end-tool}
+  ///
+  /// See also:
+  ///
+  /// * [isStylusHandwritingAvailable], which is similar, but throws an error
+  ///   when called by an unsupported API level. It directly corresponds to the
+  ///   underlying Android API
+  ///   <https://developer.android.com/reference/android/view/inputmethod/InputMethodManager#isStylusHandwritingAvailable()>.
+  ///  * [EditableText.stylusHandwritingEnabled], which controls whether
+  ///    Flutter's built-in text fields support handwriting input.
+  static Future<bool> isFeatureAvailable() async {
+    final bool? result = await _channel.invokeMethod<bool?>(
+      'Scribe.isFeatureAvailable',
+    );
+
+    if (result == null) {
+      throw FlutterError('MethodChannel.invokeMethod unexpectedly returned null.');
+    }
+
+    return result;
+  }
+
+  /// Returns true if the InputMethodManager supports Scribe stylus handwriting
+  /// input, false otherwise.
+  ///
+  /// Call this each time before calling [startStylusHandwriting] to make sure
+  /// it's available.
+  ///
+  /// Supported on Android API 34 and above. If called by an unsupported API
+  /// level, a [PlatformException] will be thrown. To avoid error handling, use
+  /// the convenience method [isFeatureAvailable] instead.
```

Comment:
Why should the user ever use this method instead of `isFeatureAvailable`?

Reply:
They could use this method if they want to distinguish between the platform not supporting Scribe (error) and the IME not supporting it (returning false). There's some discussion about it here in the engine PR and in my response that follows: https://github.com/flutter/engine/pull/52943#pullrequestreview-2378702037